src/mrt/nodejs_npm/info.rs

```diff
@@ -0,0 +1,57 @@
+use std::{path::PathBuf, fs::File, io::BufReader};
+use anyhow::{Result, Context};
+use serde::Deserialize;
+
+use crate::{package::PackageInfoExtractor};
+
+pub struct NpmPackageInfoExtractor {
+    package_json: NpmPackageJSON
+}
+
+#[derive(Deserialize, Debug)]
+struct NpmPackageJSON {
+    name: String,
+    version: String,
+}
+
+impl NpmPackageInfoExtractor {
+    pub fn from_package_path(package_path: &PathBuf) -> Result<NpmPackageInfoExtractor> {
+
+        let package_json_path = package_path.join("package.json");
+
+        let package_json_file = File::open(&package_json_path)
+            .context(format!("Reading package.json at {}", &package_json_path.display()))?;
+        let package_json_reader = BufReader::new(package_json_file);
+
+        let package_json = serde_json::from_reader(package_json_reader)
+            .context(format!("Parsing JSON of package.json at {}", package_json_path.display()))?;
+
+        return Ok(NpmPackageInfoExtractor {
+            package_json
+        });
+    }
+}
+
+impl PackageInfoExtractor for NpmPackageInfoExtractor {
+    fn get_name(&self) -> &str {
+        return self.package_json.name.as_str();
+    }
+
+    fn get_version(&self) -> &str {
+        return self.package_json.version.as_str();
+    }
+}
+
+#[test]
```

Comment:
I'm not sure that this approach scale well for your project. You definitely would have a lot more requirements and some of them would be cross-cutting.

Reply:
Not sure about what approach is your message.

Reply:
I was think that adding case which explicitly handle broken file scenario would’ve valuable.
If you are pedantic, you may add second test when package.json has merge conflicts. That’s common enough scenarios to care about

Reply:
Agree, added now have additional test.